### PR TITLE
DVDFileInfo: Fix video thumbnail extraction

### PR DESCRIPTION
**Pull request by popcornmix**
Thumbnail extraction from video files doesn't currently work.
Need to use result of GetPicture not AddData to check for a picture.

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDFileInfo.cpp [hunk 1]
```diff
@@ -242,14 +242,14 @@ bool CDVDFileInfo::ExtractThumb(const std::string &strPath,
           if (iDecoderState & VC_ERROR)
             break;
 
-          int result = 0;
-          while (result == 0)
+          iDecoderState = 0;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDFileInfo.cpp]: what does that change? looks like result is just renamed to decoderState.

Comment (popcornmix) [inline xbmc/cores/VideoPlayer/DVDFileInfo.cpp]: It means https://github.com/FernetMenta/kodi-agile/pull/93/commits/9f7703e6d7d1659122d74c140dd09eb4068e2df1#diff-d320e857fe90250ef956975e38b17212R252 uses the correct return value.

Comment (popcornmix) [inline xbmc/cores/VideoPlayer/DVDFileInfo.cpp]: (corrected that line - line 260 in the patched file)

**Comment by FernetMenta** [comment]
thanks

**Pull request merged**